kafka.go

```diff
@@ -131,12 +153,29 @@ func (kp *KafkaProducer) Produce(ctx context.Context, eventCh <-chan *events.Env
 			// Stop process immediately
 			kp.Logger.Printf("[INFO] Stop kafka producer")
 			return
+
+		case producerErr := <-kp.AsyncProducer.Errors():
+			// Instead of giving up, try to resubmit the message so that it can end up
+			// on a different partition (we don't care which partition is used)
+			// This is a workaround for https://github.com/Shopify/sarama/issues/514
+			md, _ := producerErr.Msg.Metadata.(metadata)
+			kp.Logger.Printf("[DEBUG] Got producer error %+v", producerErr)
+
+			if md.retries >= kp.repartitionMax {
+				kp.errors <- producerErr
+				continue
+			}
+
+			kp.Logger.Printf("[DEBUG] Repartioning")
+			producerErr.Msg.Metadata = metadata{retries: md.retries + 1}
```

Comment:
here it should be
```
kp.Input() <- &sarama.ProducerMessage{
 			Topic:    producerErr.Msg.Topic,
 			Value:    producerErr.Msg.Value,
 			Metadata: metadata{retries: md.retries + 1}
  		}
```
and we can drop the `producerErr.Msg.Partition = 0` (it's implicit)

This will likely require changes also to the unit tests